pypesto/objective/amici_prediction.py

```diff
@@ -0,0 +1,439 @@
+        mode:
+            Whether to compute function values or residuals.
+        """
+
+        # Do we have a maximum number of simulations allowed?
+        n_edatas = len(self.amici_objective.edatas)
+        if self.max_num_conditions == 0:
+            # simulate all conditions at once
+            n_simulations = 1
+        else:
+            # simulate only a subset of conditions
+            n_simulations = 1 + int(len(self.amici_objective.edatas) /
+                                    self.max_num_conditions)
+
+        for i_sim in range(n_simulations):
+            # slice out the conditions we actually want
+            if self.max_num_conditions == 0:
+                ids = slice(0, n_edatas)
+            else:
+                ids = slice(i_sim * self.max_num_conditions,
+                            min((i_sim + 1) * self.max_num_conditions,
+                                n_edatas))
+
+            # call amici
+            ret = self.amici_objective(x=x, sensi_orders=sensi_orders,
+                                       edatas=self.amici_objective.edatas[ids],
+                                       mode=mode, return_dict=True)
+            # post process
+            amici_t += [rdata['t'] for rdata in ret['rdatas']]
+            if 0 in sensi_orders:
+                amici_y += [rdata['y'] for rdata in ret['rdatas']]
+            if 1 in sensi_orders:
+                amici_sy += [rdata['sy'] for rdata in ret['rdatas']]
+
+    def _write_to_csv(self,
+                      outputs: Sequence[np.ndarray],
+                      outputs_sensi: Sequence[np.ndarray],
+                      timepoints: Sequence,
+                      output_file: str):
+        """
+        This method saves predictions from an amici model to a csv file.
+
+        Parameters
+        ----------
+        outputs:
+            List of postprocessed outputs (ndarrays)
+        outputs_sensi:
+            List of sensitivities of postprocessed outputs (ndarrays)
+        timepoints:
+            List of output timepoints (ndarrays)
+        output_file:
+            path to file/folder to which results will be written
+        """
+        def _prepare_csv_output(output_file):
+            """
+            If a csv is requested, this routine will create a folder for it,
+            with a suiting name: csv's are by default 2-dimensional, but the
+            output will have the format n_conditions x n_timepoints x n_outputs
+            For sensitivities, we even have x n_parameters. This makes it
+            necessary to create multiple files and hence, a folder of its own
+            makes sense.
+            """
+            # allow entering with names with and without file type endings
+            if '.' in output_file:
+                output_file_path, output_file_suffix = output_file.split('.')
+            else:
+                output_file_path = output_file
+                output_file_suffix = 'csv'
+
```

Comment:
```suggestion
            if '.' in output_file:
                output_file_path, _ = output_file.split('.')
            else:
                output_file_path = output_file
            
            output_file_suffix = 'csv'
```

Reply:
Since a `write_to_csv` function should only produce csv anyhow?

Reply:
pandas' write_csv also writes tsv's.... Well, pandas is maybe not a good reference, because I really hate its api, but this is what my code was implicitly referring to